yacs/config.py

```diff
@@ -310,66 +325,108 @@ def raise_key_rename_error(self, full_key):
     def is_new_allowed(self):
         return self.__dict__[CfgNode.NEW_ALLOWED]
 
-
-def load_cfg(cfg_file_obj_or_str):
-    """Load a cfg. Supports loading from:
-        - A file object backed by a YAML file
-        - A file object backed by a Python source file that exports an attribute
-          "cfg" that is either a dict or a CfgNode
-        - A string that can be parsed as valid YAML
-    """
-    _assert_with_logging(
-        isinstance(cfg_file_obj_or_str, _FILE_TYPES + (str,)),
-        "Expected first argument to be of type {} or {}, but it was {}".format(
-            _FILE_TYPES, str, type(cfg_file_obj_or_str)
-        ),
-    )
-    if isinstance(cfg_file_obj_or_str, str):
-        return _load_cfg_from_yaml_str(cfg_file_obj_or_str)
-    elif isinstance(cfg_file_obj_or_str, _FILE_TYPES):
-        return _load_cfg_from_file(cfg_file_obj_or_str)
-    else:
-        raise NotImplementedError("Impossible to reach here (unless there's a bug)")
-
-
-def _load_cfg_from_file(file_obj):
-    """Load a config from a YAML file or a Python source file."""
-    _, file_extension = os.path.splitext(file_obj.name)
-    if file_extension in _YAML_EXTS:
-        return _load_cfg_from_yaml_str(file_obj.read())
-    elif file_extension in _PY_EXTS:
-        return _load_cfg_py_source(file_obj.name)
-    else:
-        raise Exception(
-            "Attempt to load from an unsupported file type {}; "
-            "only {} are supported".format(file_obj, _YAML_EXTS.union(_PY_EXTS))
+    @classmethod
+    def load_cfg(cls, cfg_file_obj_or_str):
```

Comment:
This is an API breaking change. Should we maintain a module-level `load_cfg` function that basically does the following?

```python
raise NotImplementedError("load_cfg is deprecated; replace it with the class method CfgNode.load_cfg")
```

Reply:
There is still a module level one

Reply:
Ah, perfect. I missed that line on my first pass. Thanks!